src/main.cpp

```diff
@@ -84,7 +84,42 @@ void open_output_file(const std::string& a, const std::string& b) {
 
 /**************************************************************************************************/
 
-void process_orc_config_file(const char* bin_path_string) {
+template <typename T>
+T parse_enval(std::string&&) = delete;
+
+template <>
+std::string parse_enval(std::string&& x) {
```

Comment:
why move semantics instead of `const std::string&`?

Reply:
Good question. the value coming from `getenv` is a `const char*`, which is then converted to a `std::string` for this routine. For the `std::string` specialization of `parse_enval`, we can move the incoming string to the result with no additional changes. If the incoming string were `const std::string&`, it would cost us a copy to make a new one to return by value.